Pod/Classes/LSClockState.m

```diff
@@ -82,18 +86,20 @@ - (void)addSampleWithOriginMicros:(SInt64)originMicros
         latestOffsetMicros = ((receiveMicros - originMicros) + (transmitMicros - destinationMicros)) / 2;
     }
 
-    // Discard the oldest sample and push the new one.
-    [self.samples removeObjectAtIndex:0];
-    [self.samples
-        addObject:[[LSSyncSample alloc] initWithDelayMicros:latestDelayMicros offsetMicros:latestOffsetMicros]];
+    dispatch_async(self.samplesQueue, ^{
```

Comment:
I sensed you're gonna catch this one, perhaps I should write some docs.
It's a serial queue, so we keep the order the same way you'd achieve with @syncronized, NSLock or some other locking mechanism. The only difference is that you probably don't want to block the app on these writes.

i.e. if you
```
for (int i = 0; i < 3; i++) {
    dispatch_async ^{
        counter = 1;
    };
}
dispatch_sync ^{
    printf("%d", counter);
};
```
The counter will read 4. Due to the seriality of the queue, read will wait for writes to finish. Pretty much nothing different as if all of them were blocking, but you unblock the main thread from writing to disk.

If you still think it's better to do it blocking, I can easily update